tests/core/test_regions.py

```diff
@@ -613,6 +613,50 @@ def test_empty_erosion():
     assert isinstance(erosion, EmptyRegion)
 
 
+def test_intersection_sampler():
+    reg1 = BoxRegion(position=(0, 0, 0), dimensions=(0.5, 1, 1))
+    reg2 = AllRegion("all")
+    reg3 = CircularRegion((0, 0, 0), 1)
+    reg4 = reg3.footprint
+
+    regions = (reg1, reg2, reg3, reg4)
+
+    intersection_region = IntersectionRegion(*regions)
+
+    def attempt_sample(num_samples):
```

Comment:
Since this is used in 2 tests and potentially more in the future, why not move it to the top of the file as a utility function? You could call it `sample_ignoring_rejections` or something.